src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -345,9 +350,11 @@ Result AudioStreamAAudio::requestStop() {
 ResultWithValue<int32_t>   AudioStreamAAudio::write(const void *buffer,
                                      int32_t numFrames,
                                      int64_t timeoutNanoseconds) {
-    AAudioStream *stream = mAAudioStream.load();
-    if (stream != nullptr) {
-        int32_t result = mLibLoader->stream_write(mAAudioStream, buffer,
+    // Own the stream while in use.
+    std::shared_ptr<AAudioStreamOwner> mine(mAAudioStream);
```

Comment:
Does it mean that an instance of AudioStreamAAudio can be destroyed while this method is running? That can lead to bad consequences since it can be destroyed before we even get to reading of 'mAAudioStream' here. I think, a safer solution is to manage AudioStreamAAudio by a shared pointer, and hold one from all the threads that need it.